providers/flagd/src/test/java/dev/openfeature/contrib/providers/flagd/resolver/process/storage/FlagStoreTest.java

```diff
@@ -98,9 +99,9 @@ public void changedFlags() throws Exception {
         Map<String, FeatureFlag> expectedChangedFlags =
                 FlagParser.parseString(getFlagsFromResource(VALID_LONG),true);
         expectedChangedFlags.remove("myBoolFlag");
-        // flags changed from initial VALID_SIMPLE flag
-        Assert.assertEquals(expectedChangedFlags.keySet().stream().collect(Collectors.toList()),
-                storageStateDTOS.take().getChangedFlagsKeys());
+        // flags changed from initial VALID_SIMPLE flag, as a set because we don't care about order
+        Assert.assertEquals(expectedChangedFlags.keySet().stream().collect(Collectors.toSet()),
```

Comment:
This assert would intermittently fail in Java 9+ due to concurrent modification exceptions.